addons/survey/models/badge.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class GamificationBadge(models.Model):
+    _inherit = 'gamification.badge'
+
+    survey_ids = fields.One2many('survey.survey', 'certification_badge_id', 'Survey Ids')
+    survey_id = fields.Many2one('survey.survey', 'Survey', compute='_compute_survey_id', store=True)
+
+    @api.depends('survey_ids')
```

Comment:
Side note: we could improve that trigger (survey_if.certification_badge_id ?) . Otherwise we might have too much updated.